lib/util.js

```diff
@@ -772,7 +788,7 @@ function formatNamespaceObject(ctx, value, recurseTimes, keys) {
     try {
       output[i] = formatProperty(ctx, value, recurseTimes, keys[i], 0);
     } catch (err) {
-      if (!(err instanceof ReferenceError)) {
+      if (!(types.isNativeError(err) && err.name === 'ReferenceError')) {
         throw err;
```

Comment:
Could the error checks be simplified all up? If namespace objects can only error by throw for reference error then we can assume that the error accessing it is because of the reference error and remove the error checks and rethrow code above. We could also simplify the initial check by doing:

```js
try {
  keys = Object.keys(value);
} catch (err) {
  if (types.isModuleNamespaceObject(value)) {
    keys = Object.getOwnPropertyNames(value);
  } else {
    throw err;
  }
}
```
Here erroring should be rare and when it does error it is likely a namespace object so the check up-front on error should be fine.

Reply:
i think its best to explicitly check if its a reference error, do you feel strongly about this?

Reply:
I dig it if code could be simpler, and still correct, that it be simpler is all.